src/main/java/net/mtrop/doom/tools/decohack/DecoHackParser.java

```diff
@@ -1867,7 +1887,48 @@ private boolean parseStateLine(AbstractPatchContext<?> context, DEHActor actor,
 		}
 
 		state.bright = matchBrightFlag();
-		
+
+		// Maybe parse offsets
+		state.misc1 = 0;
+		state.misc2 = 0;
+		state.args.clear();
+
+		boolean useoffsets = matchOffsetDirective();
+		if( useoffsets )
+		{
+			if (matchType(DecoHackKernel.TYPE_LPAREN))
+			{
+				// no arguments
+				if (!matchType(DecoHackKernel.TYPE_RPAREN))
+				{
+					// get first argument
+					Integer p;
+					if ((p = parseActionPointerParameterValue(context, actor)) == null)
+						return false;
+					state.misc1 = p;
+				
+					if (matchType(DecoHackKernel.TYPE_COMMA))
+					{
+						if ((p = parseActionPointerParameterValue(context, actor)) == null)
+							return false;
+						state.misc2 = p;
+					}
+
+					if (!matchType(DecoHackKernel.TYPE_RPAREN))
+					{
+						addErrorMessage("Expected a ')' after offsets.");
+						return false;
+					}
```

Comment:
I think we should enforce that the use of `Offset` requires _both_ coordinates specified (X and Y offset) for user readability's sake, so I propose two changes to the code, here:

The `matchType(DecoHackKernel.TYPE_COMMA)` clause should check for `!matchType(DecoHackKernel.TYPE_COMMA)` and make an error. 

And since parsing an action pointer parameter value (via `parseActionPointerParameterValue(context, actor)`) is inappropriate here, we should use `matchInteger()` instead with an error message addition (since `matchInteger` does not add one on failure).

Reply:
Makes sense -- all done now. :)